application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadCommand.java

```diff
@@ -204,6 +205,18 @@ private void closeThread(SlashCommandInteractionEvent event, ThreadChannel helpT
         event.replyEmbeds(embed).flatMap(any -> helpThread.getManager().setArchived(true)).queue();
     }
 
+    private void bookmarkThread(SlashCommandInteractionEvent event, ThreadChannel helpThread) {
+        event.getUser()
+            .openPrivateChannel()
+            .flatMap(channel -> channel.sendMessage(String.format("<#%s>", helpThread.getIdLong())))
+            .queue();
+
+        event.reply(
+                "An attempt has made to send a link of this help thread to your DMs. Check your inbox")
+            .setEphemeral(true)
+            .queue();
```

Comment:
Make this a single action queue. with a success and failure action that have a different text.
also, use `deferReply`.

Reply:
why deferReply?

Reply:
opening a DM channel and sending a DM there, then waiting for success/failure, could take longer than 3s.

Reply:
It *could* but is very unrealistic, means you're getting heavily rate-limited or our code is really bad, I'm fine with a defer, just preferably don't want every command to defer. As it feels like it's hiding the underlying issue

Reply:
i just feel like everything that needs external resources (reading files, reading database, sending http requests, ...) has to `defer`, since it cant respond immediately.

not sure if we did that everywhere though. in particular the "reading database" bit is probably not used "correctly" everywhere

Reply:
thats fair. im okay with either for here. its more important that we have a single queue with nice error/success responses. i dont like this "i attempted sth, check ur inbox". as if the bot doesnt know whether it worked out or not - it does know.

Reply:
> Make this a single action queue. with a success and failure action that have a different text

ok but i don't think defer reply makes sense here

Reply:
🙆‍♂️